lib/bolt/transport/docker/connection.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'docker'
+require 'logging'
+require 'bolt/node/errors'
+
+module Bolt
+  module Transport
+    class Docker < Base
+      class Connection
+        def initialize(target)
+          @target = target
+          @logger = Logging.logger[target.host]
+        end
+
+        def connect
+          # Explicitly create the new Connection to avoid relying on global state in the Docker module.
+          url = @target.options['url'] || ::Docker.url
```

Comment:
I'm concerned url vs uri is going to end up being confusing here. "docker-url" "docker-api" ???

Reply:
Maybe. It'll already be
```
config:
  docker:
    url: ...
```

Reply:
What part do you feel will be confusing?

Reply:
`target.uri` vs `target.options['url']`.  Maybe `service-uri` to align with the pcp transport?

Reply:
Ok, that seems reasonable. `target.options['options']` is also probably weird.